src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java

```diff
@@ -89,7 +93,45 @@ private void addSearchPathsToClassLoader(FunctionMethodDescriptor function) thro
 
 	private void registerWithClassLoaderProvider(File libDirectory) {
 		try {
-			classLoaderProvider.addDirectory(libDirectory);
+			if(SystemUtils.IS_JAVA_1_8) {
+				String workerPath = System.getenv(Constants.FUNCTIONS_WORKER_DIRECTORY) + "/lib";
+				String reverseJarsLoading = System.getenv(Constants.JAVA_REVERSE_JARS_LOADING);
+				if(reverseJarsLoading != null && reverseJarsLoading.toLowerCase().equals("true")) {
+					if(!new File(workerPath).exists()) {
```

Comment:
Should we dramatically throw exception if we do not have access to worker lib? 
My opinion is no. But need to hear feedback.

Reply:
I talked to @fabiocav and he suggested to have it sever.